parla/task_runtime.py

```diff
@@ -1885,9 +1885,23 @@ def _schedule_tasks(self):
     def _launch_task(self, queue, dev: Device):
         try:
             task = queue.pop()
+            # Allocate additional resources used by this task (blocking)
+            for device in task.req.devices:
+                for resource, amount in task.req.resources.items():
+                    logger.debug("Task %r allocating %d %s on device %r", task, amount, resource, device)
+                self._available_resources.allocate_resources(device, task.req.resources, blocking=True)
+            # Update parray tracking and task count on the device
+            for parray in (task.dataflow.input + task.dataflow.inout + task.dataflow.output):
+                if len(task.req.environment.placement) > 1:
+                    raise NotImplementedError("Multidevice not supported")
+                for device in task.req.environment.placement:
+                    self._available_resources.register_parray_move(parray, device)
```

Comment:
This seems wrong. The data movement tasks already exist and are already run before this code executes. This is only updating the information once the "parent" task is starting to be run.

Reply:
Wait it also happens at creation, I'm not sure then. Why is it registered twice onto the same device?

Reply:
I think this should be a mistake. Then is it fine just to remove this registeration at here?

Reply:
No this is intentional. At creation its important to log the persistent memory.